### ISSUE TITLE
Add method to remove event listeners/cleanup

### ISSUE THREAD

**Issue by powerbuoy** [chartjs/chartjs-plugin-deferred#4]
I really don't mean to spam you :)

But while using the deferred plugin on my Aurelia SPA I've noticed that unless you view every chart on a page (at which point the plugin unwatches the chart's scroll event https://github.com/chartjs/chartjs-plugin-deferred/blob/master/src/plugin.js#L107); if you go to another page and scroll down you'll get `Uncaught TypeError: Cannot read property 'offsetParent' of null` from `chartInViewport()` (https://github.com/chartjs/chartjs-plugin-deferred/blob/master/src/plugin.js#L53 I believe).

I might be missing something, but is there a way to manually remove the events? That way I can do so in my ViewModel's `detached()` (or React's `componentWillUnmount()`) similarly to how I can destroy the ChartJS instance with `chart.destroy()`?

Or perhaps there's a way for the plugin itself to detect calls to `chart.destroy()` and clean up after itself?

Again, sorry if I simply missed this feature or am using the plugin incorrectly, all I've done is import it and add the `deferred` options to my chart.

**Comment by simonbrunel** [comment]
You right, that's a bug, we need to stop monitoring a chart that has been destroyed. I think it can be fixed by unwatching in the destroy extension (line [203](/chartjs/chartjs-plugin-deferred/blob/d3757dbaf2bbc7a5021dc256a7e13978532c144f/src/plugin.js#L203)):

```javascript
Chart.plugins.register({
    // ...

    beforeDatasetsUpdate: function(chart, options) {
        //...
    },

    destroy: function(chart) {
        unwatch(chart);
    }
});
```

**Comment by powerbuoy** [comment]
That sounds really promising. Is this something you could release along with `import` support?

**Comment by simonbrunel** [comment]
Sure, did you get a chance to test this fix?

**Comment by powerbuoy** [comment]
It does indeed work :) Nice job. Any ETA on a new version?

**Comment by simonbrunel** [comment]
Fixed in 8b4dacfa1193. @powerbuoy if you can confirm that this is fixed in `master`, I can trigger a new 0.3 release very soon. Just be aware that version 1.0 will contain **breaking changes** and should be released as soon as Chart.js 2.6 is out.

**Comment by powerbuoy** [comment]
Works perfectly :)

**Comment by simonbrunel** [comment]
Released in [v0.3.0](/chartjs/chartjs-plugin-deferred/releases/tag/v0.3.0)
